lightnode.go

```diff
@@ -141,52 +141,40 @@ func New(options Options, ctx context.Context, logger logrus.FieldLogger, sqlDB
 		bindings,
 	)
 
-	whitelistMap := map[tx.Selector]bool{}
-	for _, i := range options.Whitelist {
-		whitelistMap[i] = true
+	chains := map[multichain.Chain]bool{}
+	assets := map[multichain.Asset]bool{}
+	for _, selector := range options.Whitelist {
+		if !selector.IsCrossChain() {
+			continue
+		}
+		if selector.IsBurn() && selector.IsRelease() {
+			chains[selector.Source()] = true
+		}
+		assets[selector.Asset()] = true
```

Comment:
```suggestion
		if selector.IsBurn() && selector.IsRelease() {
			chains[selector.Source()] = true
			assets[selector.Asset()] = true
		}
```
I think this might be better. Unless there's a particular reason you want to leave `assets[selector.Asset()] = true` outside the if statement